src/map.js

```diff
@@ -66,99 +75,148 @@ export default class GpxMap {
                 stateName: 'default',
                 title: 'Open settings dialog',
                 onClick: (_btn, map) => {
-                    buildSettingsModal(this.options, (opts) => {
+                    buildSettingsModal(this, this.options, (opts) => {
                         this.updateOptions(opts)
-                    }).show()
-                }
-            }]
-        }).addTo(this.map)
+                    }).show();
+                },
+            }],
+        }).addTo(this.map);
+
+        let centre = this.centre.bind(this);
+
+        this.viewAll = leaflet.easyButton({
+            type: 'animate',
+            states: [{
+                icon: 'fa-map fa-lg',
+                stateName: 'default',
+                title: 'Zoom to all tracks',
+                onClick: (_btn, map) => {
+                    centre();
+                },
+            }],
+        }).addTo(this.map);
+        this.viewAll.disable();
+
+        this.switchTheme(this.options.theme);
+        this.requestBrowserLocation();
+    }
 
-        this.switchTheme(this.options.theme)
-        this.requestBrowserLocation()
+    clearScroll () {
+        this.scrolled = false;
+        this.map.addEventListener("movestart", this.markScrolled);
     }
 
     switchTheme(themeName) {
-        if (this.mapTiles)
-            this.mapTiles.removeFrom(this.map)
+        if (this.mapTiles) this.mapTiles.removeFrom(this.map);
```

Comment:
nit: I generally prefer one of 

```
if (...) {
    ...
}
```
or
```
if (...)
    ...
```

Reply:
0124b6d6505efd05507859c847895ad6f7f91c4e